docs/arch/adr-18.md

```diff
@@ -0,0 +1,47 @@
+# ADR 18: Workflow routing
+
+Created: September 4, 2019
+
+## Context
+
+On PFE, there are a couple of different ways workflows are routed:
+
+- Navigating to `/classify`, the user sees either the default workflow for that project, a random active workflow, or the workflow stored in project preferences for logged-in volunteers.
+- Navigating to `/classify?workflow=[workflow_id]` shows the workflow determined by the query parameter when the "Use workflow query" feature is enabled for the project, or if the user has the correct role (owner, collaborator, tester, or admin).
+
+This is not optimal for several reasons. Users cannot bookmark specific workflows unless they use the query parameter; there can be uncertainty over which actual workflow is being shown (particularly as the workflow name isn't shown in the UI); different workflows can be shown at the same URL at different times; and ultimately, workflows are static resources that should be routed to - that's what URLs are for.
+
+Additionally, [we currently have a requirement for CSSI CitSci to be able to route to specific subjects](https://github.com/zooniverse/front-end-monorepo/issues/806#issuecomment-495685027). It's conceivable that a project will have a requirement for routing to a specific subject set linked to a workflow as well.
+
+## Decision
+
+We adopt the following URL structure:
+
+```
+/projects/[owner]/[project]/classify/workflow/:workflow-id/subject-set/:subject-set-id/subject/:subject-id
+```
+
+This would be facilitated by the [dynamic routing feature in Next.js 9](https://github.com/zeit/next.js/#dynamic-routing) (see [#1071](https://github.com/zooniverse/front-end-monorepo/pull/1071)).
+
+Workflow links from the home page would route directly to their workflow-specific URLs.
+
+### `/classify` behaviour
+
+When navigating to `/classify`, the page should redirect the user to the correct workflow URL in the following priority:
+
+1. Workflow ID set in user project preferences by the user
```

Comment:
This is probably a paranoia question, but: do we have plans for what happens if the Workflow ID saved in the project preferences becomes invalid? e.g. that WF gets switched off by the project owner.

e.g. last week I was working on WF 666, but today the project team shut it off, because they have a new default workflow 777. (And because the old workflow kept killing all the exorcists hired to debug it.)

- Will the _"when navigating to /classify, decide where to redirect to"_ logic be the one responsible for determining that WF 666 in the user's project prefs is no longer valid, and then moving on to the next priority step?
- or will the user be greeted with a general _"sorry, invalid workflow"_ error page, preferably one with a _"if you still have trouble, pls click this to reset your user preferences"_ button?

Sorry for the slight paranoia, but I was recalling how we had reports of a user who was unable to access the Classify page of a specific project _on one single Zooniverse account,_ but was able to classify just fine anonymously or on a second Zooniverse account. We spent a very long while trying to figure out if the auto-redirect-based-on-user-project-preferences logic was the problem in that user's case.

PS: small typo, all priorities are step `1.`

Reply:
This is a multi-part question and also not entirely in scope of this ADR, but within the scope of one about workflow selection. I believe the behavior of what the UPP store does will remain the same: we don't know why the project inactivated the workflow, just that they did, so all we can do is clear it out of the UPP so that the app doesn't attempt to load this workflow every time they visit that project. 

In that scenario, what the UI does is a different question and the same as what to do when the workflow is invalid generally. I think we should show an error and ask them what they want to do. This will be the change in behavior from PFE that this ADR should note.

Reply:
Noted in 9e140f4